### PR TITLE
Fixes 'yarn start' on Windows - LavaPack is not defined (#13318)

### LINKED ISSUE #13318: [Bug]: Code in develop branch showing white screen and failing to run after build and install
### Describe the bug

I am trying to build and run the MetaMask extension on my Firefox browser, but after loading it in Firefox, it just shows a white screen. On using the browser's Inspect option, I keep getting the console errors: 'Uncaught ReferenceError: LavaPack is not defined'

image

image

### Steps to reproduce

1. `git clone` the repository: https://github.com/MetaMask/metamask-extension and checkout the branch `develop`
2. Run `yarn setup`
3. Run `yarn start`
4. After the build starts showing messages `.js` files being reloaded, open Firefox and go to `about:debugging#addons`. Click the `Load Temporary Add-on...` and select the file `metamask-extension/dist/firefox/manifest.json`
5. Once the MetaMask addon shows up in the Firefox toolbar, click it to open and use it

### Error messages or log output

```shell
Uncaught ReferenceError: LavaPack is not defined
```

### Version

develop branch, commit hash - 58ff673bcd013be809f48e9ad2bb70899d052856

### Build type

Other (please specify exactly where you obtained this build in "Additional Context" section)

### Browser

Firefox

### Operating system

Windows

### Hardware wallet

_No response_

### Additional context

_No response_

### LINKED ISSUE #13793: [Bug]:LavaPack is not defined
### Describe the bug

this error occurs on windows if you will clone it on ios or linux then metamask clone  will be  run properly. clone your metamask on Linux or Ios and follow docs of metamask

### Steps to reproduce

1.
clone code of metamask from git

install yarn if yarn not installed in your os

in terminal->

     1.yarn install
     2.yarn setup
     3.yarn start

and after that go to your chrome browser
and on developer mode and then press 'load unpacked' button 
and go to dist folder and inside dist you eill get chrome folder select that folder.
that's it.

### Error messages or log output

_No response_

### Version

10.10.1

### Build type

_No response_

### Browser

Chrome, Firefox

### Operating system

MacOS, Linux

### Hardware wallet

_No response_

### Additional context

_No response_

### LINKED ISSUE #14984: [Bug]: LavaPack is not defined
### Describe the bug

I am not sure if this is a bug or what, but I cannot run metamask locally, I have this warning error message that "LavaPack is not defined" , I have set up and then run the develop branch, BUT, when I switch to lavamoat-dev-optional, I can run the metamask locally without any problem, but the problem is that this branch is not updated and I don't have many new features.

Could someone help me out here to see what's the problem, and why these set of error messages appears on the console.

### Steps to reproduce

1.

### Error messages or log output

_No response_

### Version

develop branch

### Build type

_No response_

### Browser

Chrome

### Operating system

Windows

### Hardware wallet

_No response_

### Additional context

_No response_

### LINKED ISSUE #15237: [Bug]: Lavapack is not defined
### Describe the bug

.metamaskrc:
; Extra environment variables
PASSWORD=Develop
INFURA_PROJECT_ID=*************************** // Not 'awefoij'. Is awefoij.
SEGMENT_HOST='http://localhost:9090'
SEGMENT_WRITE_KEY='FAKE'
ONBOARDING_V2=
SWAPS_USE_DEV_APIS=
COLLECTIBLES_V1=
TOKEN_DETECTION_V2=
ADD_POPULAR_NETWORKS=

; Set this to test changes to the phishing warning page.
PHISHING_WARNING_PAGE_URL=
-----------------------------
After yarn install > yarn setup > create .metamaskrc file > yarn start > Just pause at wallet-extension\dist\brave\ui-2.js reloaded. If i load this to Chrome, i have Error Lavapack is not defined.

### Steps to reproduce

1.

### Error messages or log output

_No response_

### Version

10.17.0

### Build type

Other (please specify exactly where you obtained this build in "Additional Context" section)

### Browser

Chrome

### Operating system

Windows

### Hardware wallet

Other (please elaborate in the "Additional Context" section)

### Additional context

Скриншот 14-07-2022 170203
Скриншот 14-07-2022 170152

### PR DESCRIPTION
**Pull request by HowardBraham**
Fixes #13318, Fixes #13793, Fixes #14984, and Fixes #15237
Discussed with @kumavis and @danjm
Please let me know if this breaks another system and a more complicated fix is necessary.

Update 11/18/2022: Also adds a getPathInsideNodeModules helper function to end reliance on writing paths that start with `./node_modules/`

### CODE REVIEW

**Comment by HowardBraham** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by legobeat — DISMISSED** [review]
Can not speak for Windows. Does not break anything on Linux.

**Comment by matthewwalsh0** [comment]
Not certain if it's the same issue, but I hit a similar error building on Windows previously and resolved it by updating the copy paths before they reached `fast-glob`. 

```
async function copyGlob(baseDir, srcGlob, dest) {
    const fixedSrcGlob =
      process.platform === 'win32' ? srcGlob.replace(/\\/gu, '/') : srcGlob;

    const sources = await glob(fixedSrcGlob, { onlyFiles: false });

    await Promise.all(
      sources.map(async (src) => {
        const relativePath = path.relative(baseDir, src);
        await fs.copySync(src, `${dest}${relativePath}`);
      }),
    );
  }
 ```

File: development/build/utils.js [hunk 1]
```diff
@@ -117,10 +118,38 @@ function logError(error) {
...
+ * @param {string} pathToFiles - The path of the file or folder inside the package, optionally starting with /
+ */
+function getPathInsideNodeModules(packageName, pathToFiles) {
+  let targetPath = path.dirname(require.resolve(`${packageName}/package.json`));
+
+  if (pathToFiles.charAt(0) !== '/') {
+    targetPath += '/';
+  }
+
+  targetPath += pathToFiles;
```
Comment (legobeat) [inline development/build/utils.js:139]: Is there a reason to not do `path.join` here? Would also remove the need for manually inserting `/` above.

File: development/build/utils.js [hunk 2]
```diff
@@ -117,10 +118,38 @@ function logError(error) {
...
+
+  if (process.platform === 'win32') {
+    targetPath = targetPath.replace(/\\/gu, '/');
```
Comment (legobeat) [inline development/build/utils.js:142]: Would [`path.normalize`](https://nodejs.org/api/path.html#pathnormalizepath) be helpful here? 

Also [`path.sep`](https://nodejs.org/api/path.html#pathsep)

Comment (HowardBraham) [reply development/build/utils.js]: `path.normalize`, or even `path.posix.normalize` will not work for this.  Here's a discussion about it: https://github.com/nodejs/node/issues/12298

I replaced with the following and pushed a new commit
`targetPath.split(path.sep).join(path.posix.sep)`

File: development/build/static.js [hunk 3]
```diff
@@ -127,11 +128,14 @@ function getCopyTargets(shouldIncludeLockdown) {
...
-      src: `./node_modules/@fortawesome/fontawesome-free/webfonts/`,
+      src: getPathInsideNodeModules(
+        '@fortawesome/fontawesome-free',
+        'webfonts/',
+      ),
       dest: `fonts/fontawesome`,
     },
     {
-      src: `./node_modules/react-responsive-carousel/lib/styles`,
+      src: getPathInsideNodeModules('react-responsive-carousel', 'lib/styles'),
```
Comment (legobeat) [inline development/build/static.js:138]: above we do `images/` - any reason not to be consistent with trailing `/`?

**Review by legobeat — APPROVED** [review]
One minor thing, LGTM otherwise!

File: development/build/static.js [hunk 4]
```diff
@@ -127,11 +128,14 @@ function getCopyTargets(shouldIncludeLockdown) {
       dest: `vendor`,
     },
     {
-      src: `./node_modules/@fortawesome/fontawesome-free/webfonts/`,
+      src: getPathInsideNodeModules(
+        '@fortawesome/fontawesome-free',
```
Comment (legobeat) [inline development/build/static.js:132]:
```suggestion
        '@fontawesome/fontawesome-free',
```
typo?

**Review by legobeat — CHANGES_REQUESTED** [review]

File: development/build/static.js (hunk 4 above)
Comment (legobeat) [reply development/build/static.js:132]: Nope, nvm!

**Review by legobeat — DISMISSED** [review]

File: development/build/static.js (hunk 3 above)
Comment (HowardBraham) [reply development/build/static.js]: Yeah good call

**Review by legobeat — APPROVED** [review]

**Comment by jpuri** [comment]
I tested the PR on mac and `yarn start` works well 👍

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Pull request merged**